pyquil/operator_estimation.py

```diff
@@ -239,115 +399,83 @@ def read_json(fn):
-
-    :param op: A string representation of the Pauli operator whose +1 eigenstate we'd like to
-        prepare.
-    :param idx: The index of the qubit that the preparation is acting on
-    :return: A program which will prepare the requested state.
-    """
-    if op == 'X':
-        return Program(RY(pi / 2, idx))
-    elif op == 'Y':
-        return Program(RX(-pi / 2, idx))
-    elif op == 'Z':
+def _one_q_sic_prep(index, qubit):
+    """Prepare the index-th SIC basis state."""
+    if index == 0:
         return Program()
 
-    raise ValueError(f'Unknown operation {op}')
+    theta = 2 * np.arccos(1 / np.sqrt(3))
+    zx_plane_rotation = Program([
+        RX(-pi / 2, qubit),
+        RZ(theta - pi, qubit),
+        RX(-pi / 2, qubit),
+    ])
 
+    if index == 1:
+        return zx_plane_rotation
 
-def _local_pauli_eig_meas(op, idx):
-    """
-    Generate gate sequence to measure in the eigenbasis of a Pauli operator, assuming
-    we are only able to measure in the Z eigenbasis.
-
-    """
-    if op == 'X':
-        return Program(RY(-pi / 2, idx))
-    elif op == 'Y':
-        return Program(RX(pi / 2, idx))
-    elif op == 'Z':
-        return Program()
-    raise ValueError(f'Unknown operation {op}')
+    elif index == 2:
+        return zx_plane_rotation + RZ(-2 * pi / 3, qubit)
 
+    elif index == 3:
+        return zx_plane_rotation + RZ(2 * pi / 3, qubit)
 
-def _ops_commute(op_code1: str, op_code2: str):
-    """
-    Given two 1q op strings (I, X, Y, or Z), determine whether they commute
+    raise ValueError(f'Bad SIC index: {index}')
 
-    :param op_code1: First operation
-    :param op_code2: Second operation
-    :return: Boolean specifying whether the two operations commute
-    """
-    if op_code1 not in ['X', 'Y', 'Z', 'I']:
-        raise ValueError(f"Unknown op_code {op_code1}")
-    if op_code2 not in ['X', 'Y', 'Z', 'I']:
-        raise ValueError(f"Unknown op_code {op_code2}")
-
-    if op_code1 == op_code2:
-        # Same op
-        return True
-    elif op_code1 == 'I' or op_code2 == 'I':
-        # I commutes with everything
-        return True
-    else:
-        # Otherwise, they do not commute.
-        return False
 
+def _one_q_pauli_prep(label, index, qubit):
+    """Prepare the index-th eigenstate of the pauli operator given by label."""
+    if index not in [0, 1]:
+        raise ValueError(f'Bad Pauli index: {index}')
 
-def _all_qubits_diagonal_in_tpb(op1: PauliTerm, op2: PauliTerm):
-    """
-    Compare all qubits between two PauliTerms to see if they are all diagonal in an
-    overall shared tensor product basis. More concretely, test if ``op1`` and
-    ``op2`` are diagonal in each others' "natural" tensor product basis.
+    if label == 'X':
+        if index == 0:
+            return Program(RY(pi / 2, qubit))
+        else:
+            return Program(RY(-pi / 2, qubit))
 
-    Given some PauliTerm, the 'natural' tensor product basis (tpb) to
```

Comment:
this function is now redundant, but this was a good docstring so I put it in `group_experiments` so people might actually find it